src/components/menu/MenuConnected.tsx

```diff
@@ -0,0 +1,143 @@
+import {IReduxAction, ReduxConnect} from '../../utils/ReduxUtils';
+import {IItemBoxProps} from '../itemBox/ItemBox';
+import {ListBox} from '../listBox/ListBox';
+import {Svg} from '../svg/Svg';
+import {addMenu, removeMenu, toggleMenu} from './MenuActions';
+import {IMenuState} from './MenuReducers';
+
+export interface IMenuOwnProps {
+    items: IItemBoxProps[];
+    classes?: string[];
+    id: string;
+    svgName?: string;
+    positionRight?: boolean;
+}
+
+export interface IMenuStateProps {
+    isOpen?: boolean;
+    list?: IItemBoxProps[];
+}
+
+export interface IMenuDispatchProps {
+    onRender?: () => void;
+    onDestroy?: () => void;
+    onToggleMenu?: () => void;
+    onDocumentClick?: () => void;
+}
+
+export interface IMenuProps extends IMenuOwnProps, IMenuStateProps, IMenuDispatchProps {}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IMenuOwnProps): IMenuStateProps => {
+    const menu: IMenuState = _.findWhere(state.menus, {id: ownProps.id});
+
+    return {
+        isOpen: menu && menu.open,
+        list: (menu && menu.list) || ownProps.items,
+    };
+};
+
+const mapDispatchToProps = (
+    dispatch: (action: IReduxAction<IReduxActionsPayload>) => void,
+    ownProps: IMenuOwnProps,
+): IMenuDispatchProps => ({
+    onRender: () => dispatch(addMenu(ownProps.id, ownProps.items)),
+    onDestroy: () => dispatch(removeMenu(ownProps.id)),
+    onToggleMenu: () => dispatch(toggleMenu(ownProps.id)),
+    onDocumentClick: () => dispatch(toggleMenu(ownProps.id)),
+});
+
+@ReduxConnect(mapStateToProps, mapDispatchToProps)
+export class MenuConnected extends React.Component<IMenuProps, {}> {
+    private icon: HTMLDivElement;
+    private list: HTMLDivElement;
+
+    static defaultProps: Partial<IMenuProps> = {
+        svgName: 'more-append',
+        classes: [],
+        positionRight: false,
+    };
+
+    componentWillMount() {
+        this.props.onRender();
+        document.addEventListener('mousedown', this.handleDocumentClick);
+    }
+
+    componentWillUnmount() {
+        this.props.onDestroy();
+        document.removeEventListener('mousedown', this.handleDocumentClick);
+    }
+
+    render() {
+        const pickerClasses = classNames(
+            'select-dropdown dropdown',
+            this.props.classes,
+            {
+                open: this.props.isOpen,
+            },
+        );
+        const dropdownClasses = classNames('select-dropdown-container absolute bg-pure-white', {
+            hidden: !this.props.isOpen,
+        });
+        return (
+            <div className={pickerClasses} ref={(ref: HTMLDivElement) => this.icon = ref}>
+                <button
+                    className={classNames('btn menu-toggle', {'bg-light-grey': this.props.isOpen})}
+                    type='button'
+                    onMouseUp={(e: React.MouseEvent<HTMLElement>) => this.onToggleMenu(e)}
+                >
+                    {this.getSvg()}

```

Comment:
With my proposed change for the children you could still have a default

`{this.props.children ? this.props.children : this.getDefaultSvg()}`